zookeeper-server/src/main/java/org/apache/zookeeper/server/backup/RestoreFromBackupTool.java

```diff
@@ -328,6 +342,22 @@ public void run() throws IOException {
         throw new BackupException("Failed to create a snap directory at path: " + snapDir.getPath()
             + " to store restored snapshot files.");
       }
+      String[] dataDirFiles = dataDir.list();
+      String[] snapDirFiles = snapDir.list();
+      if (Objects.requireNonNull(dataDirFiles).length > 0
+          || Objects.requireNonNull(snapDirFiles).length > 0) {
+        if (overwrite) {
+          LOG.warn(
+              "Overwriting the destination directories for restoration. The files under dataDir: "
+                  + dataDir.getPath() + " are: " + Arrays.toString(dataDirFiles)
+                  + "; and files under snapDir: " + snapDir.getPath() + " are: " + Arrays
+                  .toString(snapDirFiles) + ".");
```

Comment:
Should we completely clear out the directory and populate it with restored files?

I'd think that if we don't clear it, restoration mostly loses its meaning because chances are, the existing files in the directory are newer than the snapshot/log files we're restoring, causing the server to mostly ignore the restored files because they are behind the existing files. Let's go back and think about what the use case for this command really is and see which behavior we should choose for the overwrite option?

Reply:
Discussed offline and reached an agreement. We will completed clear the directory and populate it with the restored files in this case. Also we will log the file names of the files that are deleted while clear out the directory. In the future, if necessary, we could consider moving the directory to a backup folder instead of clearing out the directory.